client/lib/happychat/test/index.js

```diff
@@ -79,17 +79,14 @@ describe( 'connection', () => {
 				return expect( openSocket ).resolves.toBe( socket );
 			} );
 
-			test( 'unauthorized event', () => {
+			test( 'unauthorized event', async () => {
 				socket.close = jest.fn();
-				expect.assertions( 3 );
 				socket.emit( 'unauthorized' );
-				return openSocket.catch( () => {
-					expect( dispatch ).toHaveBeenCalledTimes( 1 );
-					expect( dispatch ).toHaveBeenCalledWith(
-						receiveUnauthorized( 'User is not authorized' )
-					);
-					expect( socket.close ).toHaveBeenCalled();
-				} );
+
+				await expect( openSocket ).rejects.toBe( 'user is not authorized' );
```

Comment:
TIL `.rejects.toThrow()` can only be used if the promise was rejected with an `Error`, otherwise you should use other matches like `.reject.toBe()` to assert the rejected value.